nixos/modules/services/web-apps/mediawiki.nix

```
@@ -48,10 +48,16 @@ let
     preferLocalBuild = true;
   } ''
     mkdir -p $out/bin
-    for i in changePassword.php createAndPromote.php resetUserEmail.php userOptions.php edit.php nukePage.php update.php; do
-      makeWrapper ${php}/bin/php $out/bin/mediawiki-$(basename $i .php) \
-        --set MEDIAWIKI_CONFIG ${mediawikiConfig} \
-        --add-flags ${pkg}/share/mediawiki/maintenance/$i
+    for i in changePassword createAndPromote deleteUserEmail resetUserEmail userOptions edit nukePage update; do
+      echo "#! ${pkgs.runtimeShell}
+      sudo=exec
+      if [[ "\$USER" != ${user} ]]; then
+        sudo='exec /run/wrappers/bin/sudo -u ${user}'
```

A bit more portable, in case someone disables sudo for whatever reasons...

```suggestion
        become=(exec "${util-linux}/bin/runuser" -u "${user}" --)
```

It might be not as convenient but than it's just a `sudo` away.

---

No idea what runuser is doing with the argument parsing but that is not going to work

```
/nix/store/wsspyqpbw0sc6pgr981zwf406j7bvqp5-util-linux-2.39.4-bin/bin/runuser -u "mediawiki" -- /nix/store/6l4k41xxsqyn9kkbqfppc2xiy3zb6xx5-php-with-extensions-8.1.30/bin/php /nix/store/qzlkidc9x5xn0njr794w27xc4aww73fs-mediawiki-full-1.41.4/share/mediawiki/maintenance/run.php --conf /nix/store/s2wii1dd4y7p0bgx8pf3g2xl40n28p07-LocalSettings.php /nix/store/qzlkidc9x5xn0njr794w27xc4aww73fs-mediawiki-full-1.41.4/share/mediawiki/maintenance/resetUserEmail.php
Error: user '/nix/store/s2wii1dd4y7p0bgx8pf3g2xl40n28p07-LocalSettings.php' does not exist
```